conda_forge_tick/migrators.xsh

```diff
@@ -72,6 +83,49 @@ class PipMigrator(MiniMigrator):
             for b in self.bad_install:
                 replace_in_file(f'script: {b}', "script: {{ PYTHON }} -m pip install . --no-deps -vv", 'meta.yaml')
 
+
+class LicenseMigrator(MiniMigrator):
+    def filter(self, attrs: dict) -> bool:
+        license = attrs.get('meta_yaml', {}).get('about', {}).get('license')
+        license_fam = (attrs.get('meta_yaml', {}).get('about', {}).get('license_family', '').lower() or
+                       license.lower.partition('-')[0].partition('v')[0])
+        if license_fam in NEEDED_FAMILIES and 'license_file' not in attrs.get('meta_yaml', {}).get('about', {}):
+            return False
+        return True
+
+    def migrate(self, recipe_dir, attrs, **kwargs):
+        # Use conda build to do all the downloading/extracting bits
+        md = MetaData(recipe_dir)
+        # go into source dir
+        cb_work_dir = provide(md)
+        with indir(cb_work_dir):
+            # look for a license file
+            license_files = [(s for s in os.listdir('.') if any(s.lower().startswith(k) for k in ['license', 'copying', 'copyright']))]
+            # if there is a license file in tarball update things
+        rm -r @(cb_work_dir)
+        if license_files:
+            with indir(recipe_dir):
+                '''BSD 3-Clause License
+                  Copyright (c) 2017, Anthony Scopatz
+                  Copyright (c) 2018, The Regro Developers
+                  All rights reserved.'''
+                with open('meta.yaml', 'r') as f:
+                    raw = f.read()
+                lines = raw.splitlines()
+                ptn = re.compile(r'(\s*?)' + 'license:')
+                for i, line in enumerate(lines):
+                   m = ptn.match(line)
+                   if m is not None:
+                       break
+                ws = m.group(1)
+                if len(license_files) == 1:
+                    replace_in_file("about:", "about:\n" + ws + f"license_file: {next(iter(license_files))}", 'meta.yaml')
```

Comment:
This adds only one of the files, right?

Reply:
yes, the other clause adds the list of files.

Reply:
:smile: I read that totally wrong